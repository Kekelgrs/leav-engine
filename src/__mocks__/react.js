const react = require('react');
module.exports = {...react, memo: x => x};

export default undefined;
