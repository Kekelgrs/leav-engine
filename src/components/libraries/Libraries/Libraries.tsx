import {History} from 'history';
import React, {useState} from 'react';
import {withNamespaces, WithNamespaces} from 'react-i18next';
import {Link} from 'react-router-dom';
import {Button, Grid, Header, Icon} from 'semantic-ui-react';
import {getLibsQuery, LibrariesQuery} from '../../../queries/libraries/getLibrariesQuery';
import {addWildcardToFilters, getSysTranslationQueryLanguage} from '../../../utils/utils';
import LibrariesList from '../LibrariesList';

interface ILibrariesProps extends WithNamespaces {
    history: History;
}

function Libraries({i18n: i18next, t, history}: ILibrariesProps): JSX.Element {
    const lang = getSysTranslationQueryLanguage(i18next);
    const [filters, setFilters] = useState<any>({});

    const _onFiltersUpdate = (filterElem: any) => {
        const newElemState =
            filterElem.type === 'checkbox'
                ? filterElem.indeterminate
                    ? undefined
                    : filterElem.checked
                : filterElem.value;

        setFilters({
            ...filters,
            [filterElem.name]: newElemState
        });
    };

    return (
        <React.Fragment>
            <Grid>
                <Grid.Column textAlign="left" floated="left" width={8} verticalAlign="middle">
                    <Header size="large">
                        <Icon name="folder outline" />
                        {t('libraries.title')}
                    </Header>
                </Grid.Column>
                <Grid.Column floated="right" width={3} textAlign="right" verticalAlign="middle">
                    <Button icon labelPosition="left" size="medium" as={Link} to={'/libraries/edit'}>
                        <Icon name="plus" />
                        {t('libraries.new')}
                    </Button>
                </Grid.Column>
            </Grid>
            <LibrariesQuery query={getLibsQuery} variables={{...addWildcardToFilters(filters), lang}}>
                {({loading, error, data}) => {
                    if (typeof error !== 'undefined') {
                        return <p>Error: {error.message}</p>;
                    }

                    const onRowClick = library => history.push('/libraries/edit/' + library.id);

                    return (
                        <LibrariesList
                            loading={loading || !data}
                            libraries={data ? data.libraries : []}
                            onRowClick={onRowClick}
                            onFiltersUpdate={_onFiltersUpdate}
                            filters={filters}
                        />
                    );
                }}
            </LibrariesQuery>
        </React.Fragment>
    );
}

export default withNamespaces()(Libraries);
