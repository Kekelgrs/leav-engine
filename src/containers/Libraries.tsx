import gql from 'graphql-tag';
import * as React from 'react';
import {Query} from 'react-apollo';
import {Header, Icon} from 'semantic-ui-react';
import LibrariesList from '../components/LibrariesList';
import {GET_LIBRARIES} from '../_types/GET_LIBRARIES';

const libsQuery = gql`
    query GET_LIBRARIES {
        libraries {
            id
            label {
                fr
                en
            }
        }
    }
`;

class LibrariesQuery extends Query<GET_LIBRARIES> {}

function Libraries() {
    return (
        <div>
            <Header size="large">
                <Icon name="folder outline" />
                Libraries
            </Header>
            <LibrariesQuery query={libsQuery}>
                {({loading, error, data}) => {
                    if (loading || !data) {
                        return <p>Loading</p>;
                    }
                    if (typeof error !== 'undefined') {
                        return <p>Error: {error.message}</p>;
                    }

                    return <LibrariesList libraries={data.libraries} />;
                }}
            </LibrariesQuery>
        </div>
    );
}

export default Libraries;
