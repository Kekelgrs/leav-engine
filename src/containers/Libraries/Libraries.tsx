import {History} from 'history';
import {i18n} from 'i18next';
import * as React from 'react';
import {translate, TranslationFunction} from 'react-i18next';
import {Link} from 'react-router-dom';
import {Button, Grid, Header, Icon} from 'semantic-ui-react';
import {getSysTranslationQueryLanguage} from 'src/utils/utils';
import LibrariesList from '../../components/LibrariesList';
import {getLibsQuery, LibrariesQuery} from '../../queries/getLibrariesQuery';

interface ILibrariesProps {
    t: TranslationFunction;
    i18n: i18n;
    history: History;
}

interface ILibrariesState {
    filters: any;
}

class Libraries extends React.Component<ILibrariesProps, ILibrariesState> {
    constructor(props: ILibrariesProps) {
        super(props);

        this.state = {
            filters: {}
        };
    }

    public render() {
        const {i18n: i18next} = this.props;
        const {filters} = this.state;

        const lang = getSysTranslationQueryLanguage(i18next);

        return (
            <div>
                <Grid>
                    <Grid.Column textAlign="left" floated="left" width={8} verticalAlign="middle">
                        <Header size="large">
                            <Icon name="folder outline" />
                            {this.props.t('libraries.title')}
                        </Header>
                    </Grid.Column>
                    <Grid.Column floated="right" width={3} textAlign="right" verticalAlign="middle">
                        <Button icon labelPosition="left" size="medium" as={Link} to={'/libraries/edit'}>
                            <Icon name="plus" />
                            {this.props.t('libraries.new')}
                        </Button>
                    </Grid.Column>
                </Grid>
                <LibrariesQuery query={getLibsQuery} variables={{...filters, lang}}>
                    {({loading, error, data}) => {
                        if (typeof error !== 'undefined') {
                            return <p>Error: {error.message}</p>;
                        }

                        const onRowClick = library => this.props.history.push('/libraries/edit/' + library.id);

                        return (
                            <LibrariesList
                                loading={loading || !data}
                                libraries={data ? data.libraries : []}
                                onRowClick={onRowClick}
                                onFiltersUpdate={this._onFiltersUpdate}
                                filters={filters}
                            />
                        );
                    }}
                </LibrariesQuery>
            </div>
        );
    }

    private _onFiltersUpdate = (filterElem: any) => {
        const newElemState =
            filterElem.type === 'checkbox'
                ? filterElem.indeterminate
                    ? undefined
                    : filterElem.checked
                : filterElem.value;

        this.setState({
            filters: {
                ...this.state.filters,
                [filterElem.name]: newElemState
            }
        });
    }
}

export default translate()(Libraries);
