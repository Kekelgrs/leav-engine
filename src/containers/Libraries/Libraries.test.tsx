import * as React from 'react';
import {ApolloProvider} from 'react-apollo';
import {BrowserRouter as Router} from 'react-router-dom';
import {create} from 'react-test-renderer';
import gqlClient from '../../__mocks__/gqlClient';
import Libraries from './Libraries';

describe('Libraries', () => {
    test('Snapshot test', async () => {
        const comp = create(
            <ApolloProvider client={gqlClient}>
                <Router>
                    <Libraries />
                </Router>
            </ApolloProvider>
        );

        expect(comp).toMatchSnapshot();
    });
});
