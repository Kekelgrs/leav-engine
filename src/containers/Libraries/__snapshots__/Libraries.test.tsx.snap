// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Libraries Snapshot test 1`] = `
<div>
  <div
    className="ui grid"
  >
    <div
      className="left aligned left floated middle aligned eight wide column"
    >
      <div
        className="ui large header"
      >
        <i
          aria-hidden="true"
          className="folder outline icon"
        />
        
      </div>
    </div>
    <div
      className="right aligned right floated middle aligned three wide column"
    >
      <a
        className="ui medium icon left labeled button"
        disabled={undefined}
        href="/libraries/edit"
        onClick={[Function]}
        role="button"
        tabIndex={undefined}
      >
        <i
          aria-hidden="true"
          className="plus icon"
        />
        
      </a>
    </div>
  </div>
  <table
    className="ui selectable striped table"
  >
    <thead
      className=""
    >
      <tr
        className=""
      >
        <th
          className="one wide"
        />
        <th
          className="seven wide"
        >
          
        </th>
        <th
          className="six wide"
        >
          
        </th>
        <th
          className="one wide"
        >
          
        </th>
        <th
          className="one wide"
        />
      </tr>
      <tr
        className="filters"
      >
        <th
          className=""
        />
        <th
          className=""
        >
          <div
            className="ui small fluid input"
          >
            <input
              disabled={undefined}
              name="label"
              onChange={[Function]}
              placeholder="..."
              tabIndex={undefined}
              type="text"
              value=""
            />
          </div>
        </th>
        <th
          className=""
        >
          <div
            className="ui small fluid input"
          >
            <input
              disabled={undefined}
              name="id"
              onChange={[Function]}
              placeholder="..."
              tabIndex={undefined}
              type="text"
              value=""
            />
          </div>
        </th>
        <th
          className=""
        >
          <div
            className="ui indeterminate fitted checkbox"
            onChange={[Function]}
            onClick={[Function]}
            onMouseDown={[Function]}
          >
            <input
              checked={false}
              className="hidden"
              disabled={undefined}
              id={undefined}
              name="system"
              onClick={[Function]}
              readOnly={true}
              tabIndex={0}
              type="checkbox"
              value={undefined}
            />
            <label
              htmlFor={undefined}
            />
          </div>
        </th>
        <th
          className=""
        />
      </tr>
    </thead>
    <tbody
      className=""
    >
      <tr
        className=""
      >
        <td
          className=""
          colSpan={6}
        >
          <div
            className="ui active centered inline loader"
          >
            
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
